Sources/Delta.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+public extension IndexSet {
+
+    public static var empty: IndexSet {
+        return IndexSet()
+    }
+}
+
+public struct Delta<Snapshot: Collection, ChangeRepresentation> {
```

Comment:
Not quite fond of having the index collection type generic. Enforcing `Index == Int` or even `Indices == IndexSet` makes better sense to me.

Especially when a reactive collection is implicitly an ordered collection, it wouldn't be possible to have opaque indexes like `Dictionary` or `Set` anyway. For future reactive collection types, I don't feel like they would use anything other than `Array` or `Unsafe*Pointer` as its backing storage, and both of them work with `Int`.

Moreover, considering the adapters, you will have to require the indexes to return an `Int` so that you can convert them into `IndexPath`s. So in the end, we have an implicit constraint on the index type being convertible to `Int` without losing the order. So why not as well require `Int` anyway?

This is the lesson I've learnt when I once tried to make an abstraction over `IndexPath`.

Even if we keep it, it needs more constraints for a generic consumer to work, e.g. a constraint of `Index == Snapshot.Index`, and a `ChangeRepresentation` protocol that inherits from `RandomAccessCollection` and exposes `rangeView` of `IndexSet`.